static/js/containers/ChannelPage.js

```diff
@@ -101,20 +61,37 @@ class ChannelPage extends React.Component {
       )
     }
   }
+}
 
-  render() {
-    const { channels, postsForChannel } = this.props
-    return <Loading restStates={[channels, postsForChannel]} renderContents={this.renderContents} />
-  }
+const shouldUpdateRequirements = R.complement(R.eqProps("channelName"))
```

Comment:
For readability's sake I'd strongly prefer `R.not` to `R.complement` here. Also a brief comment here explaining the logic might help. It takes a few mental leaps (at least for me) to determine that this is going to test if the `channelName` prop has changed after a component update